autolab/Autolab
# Add rake task to clean cache

## Description
- Expire cache contents after 7 days (with `race_condition_ttl` set to 1 minute -- used to avoid race conditions)
- Add a rake task to facilitate cleaning the cache
- Update troubleshooting portion of documentation

## Motivation and Context
On 27 Oct, Autolab started returning 500 internal server errors due to the cache disk running out of inodes when attempting to write to the raw score caches. ([autolab-error.pdf](https://github.com/autolab/Autolab/files/9889510/autolab-error.pdf))

This is because FileStore caches (the default cache) are [not purged automatically](https://blog.appsignal.com/2018/04/17/rails-built-in-cache-stores.html).

Reference: https://stackoverflow.com/questions/27502082/clearing-expired-cache-entries-from-disk-cache-on-ruby-on-rails-4

Unfortunately, there does not appear to be an easy way to periodically clear expired entries. In lieu of that, this PR adds a rake task and updates the documentation to facilitate the process

## How Has This Been Tested?

**Preliminary steps**
- Enable caching in development by running `rails dev:cache`
- In `development.rb`, replace `config.cache_store = :memory_store` with `config.cache_store = :file_store, "tmp/cache/scores"`

**Test that rake task clears expired entries**
- For testing purposes, change `:expires_in` and `race_condition_ttl` to `5.seconds` (in all 3 places)
- Load the gradesheet of an assessment (e.g. AutoPopulated / Homework0), observe that the path `<AUTOLAB DIR>/tmp/cache/scores` is populated
- Wait a few seconds, run `rake user:cleanup_cache`, observe that the (expired) entries are cleared

**Test that rake task does not clear unexpired entries**
- Change `:expires_in` and `race_condition_ttl` back to their original values
- Load the gradesheet of an assessment (e.g. AutoPopulated / Homework0), observe that the path `<AUTOLAB DIR>/tmp/cache/scores` is populated
- Wait a few seconds, run `rake user:cleanup_cache`, observe that the unexpired entries are not cleared
- As a sanity check, run `rake user:clear_cache` and observe that the entries are cleared

**Documentation**
- Run docs locally (`mkdocs serve`) and navigate to `http://127.0.0.1:8000/installation/troubleshoot/#no-space-left-on-device-error`

<img width="961" alt="Screenshot 2022-12-30 at 09 39 42" src="https://user-images.githubusercontent.com/9074856/210098126-4ce32f0e-17b7-4b0e-81f7-1254971f6266.png">

## Types of changes

## Checklist:
